src/webapp/reports/glass-admin/glass-admin-list/useDeleteFiles.tsx

```diff
@@ -0,0 +1,26 @@
+import { useCallback } from "react";
+import { CompositionRoot } from "../../../../compositionRoot";
+import { Namespaces } from "../../../../data/common/clients/storage/Namespaces";
+import { useBooleanState } from "../../../utils/use-boolean";
+
+interface DeleteFilesState {
+    isDeleteModalOpen: boolean;
+    deleteFiles(ids: string[]): void;
+}
+
+export function useDeleteFiles(compositionRoot: CompositionRoot, reload: () => void): DeleteFilesState {
+    const [isDeleteModalOpen, { enable: openDeleteModal, disable: closeDeleteModal }] = useBooleanState(false);
+
+    const deleteFiles = useCallback(
+        async (ids: string[]) => {
+            openDeleteModal();
+            compositionRoot.glassAdmin
+                .updateStatus(Namespaces.FILE_UPLOADS, "delete", ids)
+                ?.then(() => closeDeleteModal());
+            reload();
+        },
```

Comment:
The function is defined as async, but no await is used. As it is, `reload` is executed without waiting fot updateState to finish, is that correct?